core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -194,10 +232,53 @@ class SYCLTeamMember {
    */
   template <typename Type>
   KOKKOS_INLINE_FUNCTION Type team_scan(const Type& value,
-                                        Type* const /*global_accum*/) const {
-    // FIXME_SYCL
-    Kokkos::abort("Not implemented!");
-    return value;
+                                        Type* const global_accum) const {
+    // We need to chunk up the whole reduction because we might not have
+    // allocated enough memory.
+    const int maximum_work_range =
+        std::min<int>(m_team_reduce_size / sizeof(Type), team_size());
+
+    int not_greater_power_of_two = 1;
+    while ((not_greater_power_of_two << 1) < maximum_work_range + 1)
+      not_greater_power_of_two <<= 1;
+
+    Type intermediate;
+    Type total{};
```

Comment:
Any reason for brace initialization for `total` and not for `intermediate`? If so, we should put a comment here explaining why, because I don't understand it off of the top of my head.

Reply:
Good catch! We don't need brace initialization for either of them.

Reply:
We still need to brace initialize `total` to get correct results so I reverted the change again. This does not hold for `intermediate`, though.